docs/pages/usage/execution.md

```diff
@@ -0,0 +1,146 @@
+---
+title: Cairo artifacts
+---
+
+# Executing cairo program to obtain execution trace, memory, and AIR inputs
+
+This document guides the user through installing and running Cairo programs to obtain execution traces, memory and AIR inputs, as well as creating and verifying tests.
+
+## Installing Rust
+
+In order to run the program you need to install Rust. Follow the instructions on the official website [here](https://www.rust-lang.org/tools/install).
+
+## Installing corelib
+
+If you have Scarb installed in your system, you can omit this first step. Since you already have the Cairo repository in your system. Otherwise follow all these steps.
+
+**1.** First you need to clone the Cairo repository:
+```bash
+git clone --depth=1 -b v2.8.0 https://github.com/starkware-libs/cairo.git
+```
+
+**2.** Second, you have to move the corelib from the cloned repository to your current path.
+```bash
+mv cairo/corelib/ .
+```
+
+**3.** Finally, remove the cloned repository:
+```bash
+rm -rf cairo/
+```
+
+## Installing cairo1-run
+To install `cairo1-run` run the following command:
+
+```bash
+cargo install --git https://github.com/lambdaclass/cairo-vm cairo1-run
+```
+
+### Installing cairo1-run from sources
+If you prefer to install it from sources you can follow these steps:
+
+Clone the repo:
+```bash
+git clone https://github.com/lambdaclass/cairo-vm.git
+```
+
+Move inside of the cairo1-run directory:
+```bash
+cd cairo-vm/cairo1-run
+```
+
+Run makefile:
+```bash
+make deps
+```
+
+
+## Running cairo1-run in a Scarb project
+As cairo1-run skips gas checks when running, you will need to add the following to your Scarb.toml to ensure that compilation is done without adding gas checks:
+
+```toml
+[cairo]
+enable-gas = false
+```
+
+Then you can compile your project with `scarb build`
+
+Then run the compiled project's sierra file located at `project_name/target/project_name.sierra.json`:
+
+```bash
+  cargo run path-to-project/target/project_name.sierra.json 
+```
+
+
+## Creating and verifying a proof of a Cairo program
+
+
+Navigate to the example test directory (`e2e_test/Cairo`):
+
+```bash
+cd e2e_test/Cairo
+```
+
+Compile and run the program to generate the prover input files:
+
+```bash
+cargo run ../../fibonacci.cairo \
```

Comment:
If we installed cairo1-run, we can use it instead of cargo run, like that
```
cairo1-run fibonacci.cairo     --layout=small     --air_public_input=fibonacci_public_input.json     --air_private_input=fibonacci_private_input.json     --trace_file=fibonacci_trace.bin     --memory_file=fibonacci_memory.bin     --proof_mode
```